File: src/programmers/p1/P1_12915.java
package programmers.p1;

import java.util.Arrays;
import java.util.Comparator;

public class P1_12915 {
    public static void main(String[] args) {
        System.out.println(Arrays.toString(solution(new String[]{"sun", "bed", "car"}, 1)));
        System.out.println(Arrays.toString((solution(new String[]{"abce", "abcd", "cdx"}, 2))));
    }

    static public String[] solution(String[] strings, int n) {
        return Arrays.stream(strings).sorted().sorted(Comparator.comparing(it -> it.charAt(n))).toArray(String[]::new);
    }
}

File: src/Programmers/P1/P1_12915.kt
package Programmers

import java.util.*

fun main() {
    println(solution(arrayOf("sun", "bed", "car"), 1).contentToString())
    println(solution(arrayOf("abce", "abcd", "cdx"), 2).contentToString())
}

private fun solution(strings: Array<String>, n: Int): Array<String> {
    strings.also { it ->
        it.sort()
        it.sortBy { it[n] }
    }
    return strings
}

private fun solution2(strings: Array<String>, n: Int): Array<String> {
    Arrays.sort(strings)
    for (i in strings.indices) {
        strings[i] = strings[i][n] + strings[i]
    }
    Arrays.sort(strings)
    for (i in strings.indices) {
        strings[i] = strings[i].substring(1)
    }
    return strings
}